src/onnx/parse_matmul.cpp

```diff
@@ -148,6 +206,108 @@ struct parse_matmul : op_parser<parse_matmul>
         }
     }
 
+    static void handle_scaled_transposes(const onnx_parser::node_info& info,
+                                         instruction_ref& scale_a0,
+                                         instruction_ref& zp_a0,
+                                         bool no_zp)
+    {
+        if(no_zp)
```

Comment:
It seems only `zp_a0` needs to be in the if clause..?

Reply:
No the second input as its bound by column of the input vector (which is 1-d always). 

I broke this out instead of adding it inline to encapsulate logic.

Reply:
renamed it